### PR TITLE
chore: build.yml file is updated for more strict frontend checks

### LINKED ISSUE #905: update build.yml for strict frontend checks before release
## Is your feature request related to a problem?

*Please describe.*
right now pipeline allow error over eslint

## Describe the solution you'd like
blocking the build when eslint error is present

### PR DESCRIPTION
**Pull request by palash-signoz**
closes #905

### CODE REVIEW

**Review by pranshuchittora — DISMISSED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ on:
       - main
       - v*
```
[the remark this answers is not in the document]
Comment (palash-signoz) [reply .github/workflows/build.yaml:6]: can we take this one up .. in another issue as this was not present earlier ? _(with Axolo)_

[the remark this answers is not in the document]
Comment (palash-signoz) [reply .github/workflows/build.yaml:6]: updated @prashant-shahi

**Review by prashant-shahi — APPROVED** [review]
:shipit: LGTM

**Review by pranshuchittora — APPROVED** [review]

**Review by pranshuchittora — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -39,12 +40,14 @@ jobs:
...
-        continue-on-error: true
       - name: Run ESLint
         run: cd frontend && npm run lint
-        continue-on-error: true
+      - name: TSC
+        run: yarn tsc
+        working-directory: ./frontend
+      - name: Build - Webpack
+        run: yarn run build
+        working-directory: ./frontend
```
Comment (prashant-shahi) [inline .github/workflows/build.yaml:50]: `make build-frontend-amd64`  runs the `yarn build` so it is duplicated.

Do checkout the commands the Dockerfile below to verify the same:
https://github.com/SigNoz/signoz/blob/develop/frontend/Dockerfile

Comment (palash-signoz) [reply .github/workflows/build.yaml]: aah my bad ... please re-review the code one more time i have removed the duplicate steps

**Review by prashant-shahi — APPROVED** [review]

**Review by ankitnayan — APPROVED** [review]

**Pull request merged**